vllm/sequence.py

```diff
@@ -634,6 +647,8 @@ def __init__(
         self.computed_block_nums = computed_block_nums
         self.multi_modal_data = multi_modal_data
         self.state = SequenceGroupState() if state is None else state
+        self.encoder_seq_data = encoder_seq_data
```

Comment:
If encoder_seq_data is set what is the expected value of prompt_token_ids included in seq_data? Is it bos_token_id? If so wondering if we can add some validations here to verify that.

Reply:
See my response here https://github.com/vllm-project/vllm/pull/4837#discussion_r1608201961

I will resolve this conversation for now, but feel free to reopen it if necessary.